### PR TITLE
[Feature Store] Make `name` optional in `featureset.add_aggregation`

### PR DESCRIPTION
**Pull request by katyakats**

### CODE REVIEW

File: mlrun/feature_store/feature_set.py [hunk 1]
```diff
@@ -453,13 +454,40 @@ def graph(self):
...
+        return True
+
     def add_aggregation(
         self,
-        name,
         column,
         operations,
         windows,
         period=None,
+        name=None,
```
Comment (theSaarco) [inline mlrun/feature_store/feature_set.py:490]: Why not put it either in the end of params or as the 1st optional parameter? Any reason why this is placed here?

File: tests/system/feature_store/test_feature_store.py [hunk 2]
```diff
@@ -104,9 +104,27 @@ def _ingest_quotes_featureset(self):
...
+            assert False
+        except mlrun.errors.MLRunInvalidArgumentError:
+            pass
+
+        try:
+            # no name parameter, different period
+            quotes_set.add_aggregation("bid", ["max"], "1h", "5m")
+            assert False
+        except mlrun.errors.MLRunInvalidArgumentError:
+            pass
```
Comment (theSaarco) [inline tests/system/feature_store/test_feature_store.py:123]:
```suggestion
        with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
            # no name parameter, different period
            quotes_set.add_aggregation("bid", ["max"], "1h", "5m")
```

This is more correct - it ensures we're getting the right exception.

File: tests/system/feature_store/test_feature_store.py [hunk 3]
```diff
@@ -104,9 +104,27 @@ def _ingest_quotes_featureset(self):
...
         )
 
-        quotes_set.add_aggregation("asks1", "ask", ["sum", "max"], "1h", "10m")
-        quotes_set.add_aggregation("asks2", "ask", ["sum", "max"], "5h", "10m")
-        quotes_set.add_aggregation("bids", "bid", ["min", "max"], "1h", "10m")
+        quotes_set.add_aggregation("ask", ["sum", "max"], "1h", "10m", name="asks1")
+        quotes_set.add_aggregation("ask", ["sum", "max"], "5h", "10m", name="asks2")
+        quotes_set.add_aggregation("bid", ["min"], "1h", "10m")
+
+        try:
```
Comment (theSaarco) [inline tests/system/feature_store/test_feature_store.py:111]: See next comment - use `pytest.raises()`

**Review by theSaarco — CHANGES_REQUESTED** [review]
I have mostly cosmetic comments, there's one doubt I have on the logic you added - please see the comments.

File: tests/system/feature_store/test_feature_store.py [hunk 4]
```diff
@@ -104,9 +104,21 @@ def _ingest_quotes_featureset(self):
...
+        quotes_set.add_aggregation("ask", ["sum", "max"], "5h", "10m", name="asks2")
+        quotes_set.add_aggregation("bid", ["min"], "1h", "10m")
+
+        with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+            # no name parameter, different period
+            quotes_set.add_aggregation("bid", ["max"], "1h", "5m")
+
+        with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+            # no name parameter, different period
+            quotes_set.add_aggregation("bid", ["max"], "1h", "5m")
```
Comment (theSaarco) [inline tests/system/feature_store/test_feature_store.py:117]: Now you have this same test twice - I believe you meant to do the 2nd test with a different window?

**Review by theSaarco — APPROVED** [review]
LGTM

**Comment by katyakats** [comment]
@theSaarco approved this with Shahar

**Pull request merged**